x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/transforms/DataFrameTransformTask.java

```diff
@@ -240,7 +245,16 @@ public void setTaskStateStopped() {
      *                           current checkpoint is not set
      * @param listener Started listener
      */
-    public synchronized void start(Long startingCheckpoint, ActionListener<Response> listener) {
+    public synchronized void start(Long startingCheckpoint, boolean force, ActionListener<Response> listener) {
```

Comment:
I opted for moving the `force` check (if we end up getting past our earlier checks against the cluster state) INTO these synchronized methods. This gives us some ordering guarantees that make these state transitions easier to reason about.